api4/channel.go

```diff
@@ -17,25 +17,25 @@ import (
 )
 
 func (api *API) InitChannel() {
-	api.BaseRoutes.Channels.Handle("", api.APISessionRequired(getAllChannels)).Methods("GET")
-	api.BaseRoutes.Channels.Handle("", api.APISessionRequired(createChannel)).Methods("POST")
-	api.BaseRoutes.Channels.Handle("/direct", api.APISessionRequired(createDirectChannel)).Methods("POST")
-	api.BaseRoutes.Channels.Handle("/search", api.APISessionRequiredDisableWhenBusy(searchAllChannels)).Methods("POST")
+	api.BaseRoutes.Channels.Handle("", api.APISessionRequired(getAllChannels, model.ScopeChannelsRead)).Methods("GET")
+	api.BaseRoutes.Channels.Handle("", api.APISessionRequired(createChannel, model.ScopeChannelsCreate)).Methods("POST")
+	api.BaseRoutes.Channels.Handle("/direct", api.APISessionRequired(createDirectChannel, model.ScopeChannelsCreate)).Methods("POST")
+	api.BaseRoutes.Channels.Handle("/search", api.APISessionRequiredDisableWhenBusy(searchAllChannels, model.ScopeChannelsSearch)).Methods("POST")
```

Comment:
I think this was briefly discussed already but I'd be curious to hear your thoughts on requiring just `ScopeChannelsSearch` versus `ScopeChannelsSearch`, `ScopeChannelsRead`, and even `ScopeTeamsRead` (since the API returns `ChannelListWithTeamData`)? What's the rationale behind not requiring the read scopes?

I don't have a strong opinion on this in either direction, just looking to understand the decision.

Reply:
1/5 @jupenur @asatkinson the intended audience for Scopes are sysadmins (and ultimately, end-users) that consent to installing an app. "Search" may be perceived as a different type of feature - in terms of its information "leakiness", and system performance impacts.

After some trimming, here's the current mappings of the exposed search APIs,
```go
			"searchAllChannels":                  {model.ScopeChannelsSearch},
			"searchChannelsForTeam":              {model.ScopeChannelsSearch, model.ScopeTeamsRead},
			"searchFilesForUser":                 {model.ScopeFilesSearch, model.ScopeUsersRead},
			"searchFilesInTeam":                  {model.ScopeFilesSearch, model.ScopeTeamsRead},
			"searchPostsInAllTeams":              {model.ScopePostsSearch},
			"searchPostsInTeam":                  {model.ScopePostsSearch, model.ScopeTeamsRead},
			"searchTeams":                        {model.ScopeTeamsSearch},
			"searchUsers":                        {model.ScopeUsersSearch},
```
`SearchX...ForY` APIs require `Y:read` since they need to access a specific resource of kind Y, otherwise only `X:search` is needed, even if returns details normally requiring `X:read`.

Reply:
Basically, "x:search" is inclusive of "x:read" for the purpose of the search APIs, but not of any "other:read". I am also 0/5, I don't see particular harm in requiring "x:read" explicitly, but seems a little redundant/wordy.

Reply:
> "Search" may be perceived as a different type of feature - in terms of its information "leakiness", and system performance impacts.

This makes sense...

> `SearchX...ForY` APIs require `Y:read` since they need to access a specific resource of kind Y, otherwise only `X:search` is needed, even if returns details normally requiring `X:read`.

...as does this...

> Basically, "x:search" is inclusive of "x:read" for the purpose of the search APIs, but not of any "other:read".

...and this.

But shouldn't that then imply that the correct scopes for `searchAllChannels` are `ScopeChannelsSearch` and `ScopeTeamsRead`, as per my original question?